cfme/tests/networks/nuage/test_operations.py

```diff
@@ -0,0 +1,39 @@
+import pytest
+
+from cfme.utils.log import logger
+from cfme.utils.wait import wait_for
+from wrapanapi.utils.random import random_name
+from cfme.networks.provider.nuage import NuageProvider
+
+pytestmark = [
+    pytest.mark.provider([NuageProvider])
+]
+
+
+def test_router_add_subnet(provider, with_nuage_sandbox):
+    """
+    Ensure that subnet is added on network router
+
+    We navigate to router through Provider > Tenant > Network Router
+    """
+    sandbox = with_nuage_sandbox
+    tenant_name = sandbox['enterprise'].name
+    router_name = sandbox['domain'].name
+    tenant = provider.collections.cloud_tenants.instantiate(name=tenant_name, provider=provider)
+    router = tenant.collections.routers.instantiate(name=router_name)
+    subnet_name = random_name()
```

Comment:
Optional:
I would you `fauxfactory` to create random name.

Reply:
Function `random_name()` is imported from the other file, where it is calling `fauxfactory` to generate a random name :).

Reply:
Noted @aljazkosir but wrapanpi could change its random_name method, as its a util for use with wrapanpi, not necessarily for direct use in test. It could even just change the default to something incompatible with the subnet name field. Using fauxfactory directly allows you to control why type (alphanumeric, numeric, alpha) and length directly.